File: jvm-experiments-timeperiod/src/main/java/kr/kth/timeperiod/IntervalEdge.java
package kr.kth.timeperiod;

/**
 * {@link ITimeInterval} 경계의 상태 (개구간, 폐구간)
 * User: sunghyouk.bae@gmail.com
 * Date: 12. 9. 18.
 */
public enum IntervalEdge {
    /**
     * 폐구간
     */
    Closed,

    /**
     * 개구간
     */
    Open
}

File: jvm-experiments-timeperiod/src/main/scala/kr/kth/timeperiod/IntervalEdge.scala
package kr.kth.timeperiod

/**
 * 구간의 끝의 열림/닫힘을 구분합니다.
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 13.
 */
object IntervalEdge extends Enumeration {

  type IntervalEdge = Value

  // NOTE: withName 메소드를 사용하세요

  /**
   * 페구간
   */
  val Closed = Value("Closed")
  /**
   * 개구간
   */
  val Open = Value("Open")
}
